src/crystal/static_list.cr

```diff
@@ -0,0 +1,98 @@
+# :nodoc:
+macro container_of(ptr, type, member)
```

Comment:
Even though this is `:nodoc:` it leaks to the top-level namespace. So in my code I can call `container_of` and it will call this macro. This should probably be moved inside `Crystal::StaticList` and called like `Crystal::StaticList.container_of`. Same for the other macro.

Reply:
OK, I will move these two macros inside the struct. Thank you.

Reply:
@asterite Sorry for disturbing.  I have a problem when I put them into `Crystal::StaticList`. The macros seem to be unable to know the `Sender` and `Receiver` because of visibility. And now I find three ways to resolve it.

1. Remove the `private` of `Sender` and `Receiver` so that it can call `Crystal::StaticList.container_of`. But the problem is it will leak `Sender` and `Receiver` to public.

2. Pack macros into a module like `Crystal::StaticList::Reflect`, and include the module into channel. It will call the` container_of` instead of `Crystal::StaticList.container_of`. And this way will be ugly and hard to understand.

3. Use annotation and macro to generate corresponding method. Then it will invoke
something like `Receiver(T).from_{{ member_name }}(ptr)`. This could increase the compile time because of extra computation.

Here is the code pieces about these changes
https://gist.github.com/firejox/a6f2193a5e0820eef7e7e36c937c2330

Maybe there are other clever ways that I didn't come out. What do you think?

Reply:
Instead of private visibility you can just put :nodoc: in a doc comment.

Reply:
@asterite IMO they shouldn't leak into the top-level scope anyway.

Reply:
OK, I will use the :nodoc: . Thank you.